TombEngine/Objects/TR5/Entity/tr5_lion.cpp

```diff
@@ -118,94 +142,98 @@ namespace TEN::Entities::Creatures::TR5
 						break;
 					}
 
-					if (AI.ahead)
+					if (targetAhead && AI.bite)
 					{
-						if (item->TouchBits.Test(LionAttackJoints))
-						{
-							item->Animation.TargetState = LION_STATE_BITE_ATTACK;
-							break;
-						}
-
-						if (AI.distance < LION_POUNCE_ATTACK_RANGE)
+						if (AI.distance >= LION_ATTACK_MIN_RANGE)
 						{
-							item->Animation.TargetState = LION_STATE_POUNCE_ATTACK;
-							break;
+							if (AI.distance <= LION_BITE_ATTACK_RANGE)
+							{
+								item->Animation.TargetState = LION_STATE_BITE_ATTACK;
+								break;
```

Comment:
you can remove this break;

Reply:
As it's right now, no I can't.

Because if I quit that break, the thread will continue and will replace TargetState with the running state of line 162.

This lion, although code is very similar to the other entities of TEN, still has a change in the pattern.

The other creatures do

```
Behavior case: IDLE
   if (this)
     foo1
   else if (this 2)
      foo2
   else if (this 3)
      foo3
   else
      foo4
```
In that case I would omit the middle breaks because I know when any of the foo happens, the rest of the code won't happen.

But lion does

```
Behaviour case IDLE
 if (this)
     foo1 
 if (this2)
      foo2
 if (this3)
      foo3
foo4

```
Here I can't omit because if a foo happens, that's don't garanteed the next ones don't happen.


I though in re-write all the lion code to make it like the other entities, but I though it could be any fixation of mine, so I respected the structure no "else if" as wrote... whoever wrote it. O_ò

Reply:
Okay no problem then